lib/Trash/TrashBackend.php

```diff
@@ -224,26 +237,24 @@ private function unwrapJails(IStorage $storage, string $internalPath): array {
 		return [$unJailedStorage, $unJailedInternalPath];
 	}
 
-	private function userHasAccessToFolder(IUser $user, int $folderId) {
+	private function userHasAccessToFolder(IUser $user, int $folderId): bool {
 		$folders = $this->folderManager->getFoldersForUser($user);
 		$folderIds = array_map(function (array $folder) {
 			return $folder['folder_id'];
 		}, $folders);
 		return in_array($folderId, $folderIds);
 	}
 
-	private function userHasAccessToPath(IUser $user, string $path, $permission = Constants::PERMISSION_READ) {
+	/**
+	 * @psalm-param 1|2|8 $permission
```

Comment:
Isn’t permission a mask that may be an aggregate of permissions?
Meaning it may be 3 for read&write for instance.

Reply:
Psaml auto-fix tried to be clever :) Fixed now